Code/GraphMol/ChemReactions/Reaction.cpp

```diff
@@ -646,4 +646,18 @@ void ChemicalReaction::removeUnmappedProductTemplates(
   res_productTemplates.clear();
 }
 
+void ChemicalReaction::removeAgentTemplates(
+    MOL_SPTR_VECT *targetVector)
+{
+  if(targetVector){
```

Comment:
that's only the case if targetVector is empty on function call.

It would be shorter to write this as: `std::insert(targetVector->end(),beginAgentTemplates(),endAgentTemplates())` (untested)